commands/verify-policy/verify-policy.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/CheckPointSW/terraform-provider-checkpoint/commands"
+)
+
+type arrayFlags []string
+
+func (i *arrayFlags) String() string {
+	return "my string representation"
+}
+
+func (i *arrayFlags) Set(value string) error {
+	*i = append(*i, value)
+	return nil
+}
+
+var targets arrayFlags

```

Comment:
Why you need arrayFlags? I think it's left overs from the other scripts

Reply:
That is used to parse the flags see also : https://stackoverflow.com/questions/28322997/how-to-get-a-list-of-values-into-a-flag-in-golang

By using:
`import "flags"`
You still need to define the `arrayFlags` to be able to use the flags in the main function
```
func main() {
    flag.Var(&myFlags, "list1", "Some description for this param.")
    flag.Parse()
}
```

Reply:
We don't need list of values in this command just string argument for policy package.
I also did not see flag parser use it in your code e.g. no usage of flag.Var(..) function which is ok because we don't have list of arguments flag.
Did you test your script?

Reply:
Yes I compiled it and it does trigger a "verify_policy" on my local Checkpoint Security Manager.

As for the `flag.var()` that is what this does right
`flag.StringVar(&policyPackage, "policy-package", "", "The name of the Policy Package to be installed.")`

Reply:
Ok good. Can you post test results / picture of command execution on your local environment?
Please notice that flag.StringVar() is not the same as flag.Var(). These functions accept different parameters and that's why 'arrayFlags' implementation is unnecessary here. Please remove it from the code.

Reply:
roel@LAPTOP-HNOPE3SA:/mnt/c/Git/Ahold/ahit-cft-network-checkpoint-tf [Feature/update_auto-tfvars !?]$ terraform apply --auto-approve && cp_publish && verify_policy -policy-package "standard"
module.cp_service_tcp_object["TCP_1540"].checkpoint_management_service_tcp.main: Refreshing state... [id=0cf19fb9-90d0-46c3-92a6-a45b12a5f537]
module.cp_service_tcp_object["TCP_61717"].checkpoint_management_service_tcp.main: Refreshing state... [id=dcf07791-887c-47ee-8785-d536988580e9]
module.cp_service_tcp_object["TCP_61716"].checkpoint_management_service_tcp.main: Refreshing state... [id=8855286b-a685-4f5b-b33f-1e4c30ee462f]
module.cp_host_object["host_10.10.10.1"].checkpoint_management_host.main: Refreshing state... [id=29595cea-4360-467d-8e83-e9cc87284b8a]
module.cp_host_object["host_100.10.20.1"].checkpoint_management_host.main: Refreshing state... [id=16ee715d-741a-4303-b911-1a5f05b6a981]
checkpoint_management_access_rule.rule1: Refreshing state... [id=95f9684f-536a-40b9-a174-5ac339fa22d6]
module.cp_service_udp_object["UDP_23"].checkpoint_management_service_udp.main: Refreshing state... [id=f0e96fb3-8d67-43e8-9b9a-ec083cad8549]
module.cp_security_section["Northbound Outbound Traffic"].checkpoint_management_access_section.main: Refreshing state... [id=96f0c397-8197-4a62-9905-f3dccaebcbd4]
module.cp_security_section["Cleanup"].checkpoint_management_access_section.main: Refreshing state... [id=489f1a5d-1007-4c85-b70d-555382c8acd6]
module.cp_security_section["Southbound Outbound Traffic"].checkpoint_management_access_section.main: Refreshing state... [id=b81b4c17-159a-4902-aba0-16af2fbda8e1]
module.cp_security_section["Original rules"].checkpoint_management_access_section.main: Refreshing state... [id=99937433-a4d1-4e3b-b46e-1363c40e9364]
module.cp_security_section["Southbound Inbound Traffic"].checkpoint_management_access_section.main: Refreshing state... [id=bea2cb89-7ef8-4b20-8ce8-e906f7b6774c]
module.cp_network_object["net_100.10.22.0_24"].checkpoint_management_network.main: Refreshing state... [id=2d9b6e30-f12c-47fa-a7c0-218e50057326]
module.cp_network_object["net_10.10.27.0_24"].checkpoint_management_network.main: Refreshing state... [id=59d548e1-0186-42bb-a734-1ade3995eead]
module.cp_network_object["net_10.10.37.0_24"].checkpoint_management_network.main: Refreshing state... [id=1e7a7aa4-49fe-4748-ae2f-d87f2b3c2421]
module.cp_network_object["net_10.10.36.0_24"].checkpoint_management_network.main: Refreshing state... [id=0ed8b642-84f3-47ef-8f9a-b036388c230d]
module.cp_network_object["net_10.10.22.0_24"].checkpoint_management_network.main: Refreshing state... [id=aea11cd5-5c5e-46b3-88e8-285498e889b2]
module.cp_network_object["net_10.10.39.0_24"].checkpoint_management_network.main: Refreshing state... [id=0508c720-bf43-484b-b581-699433deaf9b]
module.cp_network_object["AHO-141.93.177.222_32"].checkpoint_management_network.main: Refreshing state... [id=375d57b8-ba4f-4345-8f9b-740d8258455e]
module.cp_network_object["AHO-10.72.72.16_28"].checkpoint_management_network.main: Refreshing state... [id=7e47f338-1318-40a8-b714-a53fb6c848d1]
module.cp_network_object["net_10.10.26.0_24"].checkpoint_management_network.main: Refreshing state... [id=184169c0-0fe6-47d0-8ca6-bd2dd40a8951]
module.cp_network_object["net_10.10.20.0_24"].checkpoint_management_network.main: Refreshing state... [id=99ac786f-6e54-428c-8cda-3741d3c567fd]
module.cp_network_object["AHO-141.93.125.142_32"].checkpoint_management_network.main: Refreshing state... [id=121ea4c3-830b-4d0b-a0ee-be64b4105213]
module.cp_network_object["net_10.10.24.0_24"].checkpoint_management_network.main: Refreshing state... [id=430b05ad-9c3c-4aef-a338-0fb38df00efb]
module.cp_network_object["net_10.10.32.0_24"].checkpoint_management_network.main: Refreshing state... [id=b0e146f1-d2af-450c-9966-5fbc58276449]
module.cp_network_object["net_10.10.30.0_24"].checkpoint_management_network.main: Refreshing state... [id=ea6846b8-5a06-415f-8cfc-cc0f17ffe231]
module.cp_network_object["AHO-10.72.8.80_28"].checkpoint_management_network.main: Refreshing state... [id=a15d4a68-d8ed-4af3-8c60-0fa7ccbb0db0]
module.cp_network_object["AHO-141.93.177.220_32"].checkpoint_management_network.main: Refreshing state... [id=9c9a5071-76f8-4424-a16b-f0e7b31480c9]
module.cp_network_object["net_10.10.10.0_24"].checkpoint_management_network.main: Refreshing state... [id=8ebe3cc5-62fd-4eec-9171-a95fbfaa080e]
module.cp_network_object["net_10.10.23.0_24"].checkpoint_management_network.main: Refreshing state... [id=d7461c11-02c4-46ae-bd4d-cb73fbbc5cf3]
module.cp_network_object["net_10.10.28.0_24"].checkpoint_management_network.main: Refreshing state... [id=7ce44dd3-1032-403a-9387-0924fdc6294a]
module.cp_network_object["net_10.10.34.0_24"].checkpoint_management_network.main: Refreshing state... [id=334f753d-1f97-437a-a9c1-b089ec077357]
module.cp_network_object["net_100.10.10.0_24"].checkpoint_management_network.main: Refreshing state... [id=affc92f3-1090-4e60-a708-ee00c94558f8]
module.cp_network_object["net_10.10.33.0_24"].checkpoint_management_network.main: Refreshing state... [id=e3bd53f5-640d-4dcc-912f-ac7975c37cb2]
module.cp_network_object["net_10.10.21.0_24"].checkpoint_management_network.main: Refreshing state... [id=4ef622b8-75c3-4205-ad48-0a984354edcf]
module.cp_network_object["net_10.10.29.0_24"].checkpoint_management_network.main: Refreshing state... [id=1761ba9e-8c2d-4766-8959-37500453519d]
module.cp_network_object["AHO-10.73.8.80_28"].checkpoint_management_network.main: Refreshing state... [id=9304562c-4606-4175-b4c3-b868dd22dddb]
module.cp_network_object["net_10.10.31.0_24"].checkpoint_management_network.main: Refreshing state... [id=9c476f46-d8aa-4631-b4ea-d080b71a17a1]
module.cp_network_object["net_100.10.21.0_24"].checkpoint_management_network.main: Refreshing state... [id=f1d7020b-0f3a-4cf6-98a4-6d1d9cc9ed04]
module.cp_network_object["net_10.10.38.0_24"].checkpoint_management_network.main: Refreshing state... [id=6da3571c-45c1-4ab1-a282-ab505c25b934]
module.cp_network_object["net_100.10.20.0_24"].checkpoint_management_network.main: Refreshing state... [id=d188d5fd-032f-438d-a934-41007626deec]
module.cp_network_object["AHO-10.73.72.16_28"].checkpoint_management_network.main: Refreshing state... [id=228b9e00-f983-4ee7-860f-9bfda7d15461]
module.cp_network_object["net_10.10.35.0_24"].checkpoint_management_network.main: Refreshing state... [id=0d66657c-c8be-4d33-91f3-907930003a7d]
module.cp_network_object["AHO-141.93.125.141_32"].checkpoint_management_network.main: Refreshing state... [id=443b7647-e131-4593-8e88-abde94682893]
module.cp_network_object["net_10.10.25.0_24"].checkpoint_management_network.main: Refreshing state... [id=81508224-bd3f-4275-b11c-b28fab84f90d]
module.cp_group_object["group_3"].checkpoint_management_group.main: Refreshing state... [id=481e4ec7-5962-491f-86eb-f147ff87c274]
module.cp_group_object["group_1"].checkpoint_management_group.main: Refreshing state... [id=28771903-5524-41d4-a75f-8e62c1698c39]
module.cp_group_object["group_2"].checkpoint_management_group.main: Refreshing state... [id=d0cc1a01-b6f2-4a8f-82be-f5af7c2c8a82]
module.cp_group_object["NoEu_STH_Gateways"].checkpoint_management_group.main: Refreshing state... [id=29a91117-deb3-446b-8fad-b60f5c8ba8fa]
module.cp_access_rule["test4"].checkpoint_management_access_rule.main: Refreshing state... [id=212cef81-c7af-4ef1-9d50-2513edbcaaea]
module.cp_access_rule["test5"].checkpoint_management_access_rule.main: Refreshing state... [id=fe63944e-bfec-477b-ac72-77306703b1c7]
module.cp_access_rule["test2"].checkpoint_management_access_rule.main: Refreshing state... [id=55a02958-3908-4ae9-ba17-f2ce93e4ac77]
module.cp_access_rule["test3"].checkpoint_management_access_rule.main: Refreshing state... [id=263cf225-dd75-483a-ad5b-b2982120a350]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
Publish finished successfully. task-id [01234567-89ab-cdef-96e5-1ea28ecad982]
**Verify finished successfully. task-id [01234567-89ab-cdef-b191-0c5bb8a26027]**

![image](https://user-images.githubusercontent.com/16721844/219362861-f653a4b6-a8ab-4604-9e86-c3992e2def7b.png)

Reply:
> Ok good. Can you post test results / picture of command execution on your local environment? Please notice that flag.StringVar() is not the same as flag.Var(). These functions accept different parameters and that's why 'arrayFlags' implementation is unnecessary here. Please remove it from the code and see script run as expected.

I think I over reached my knowledge of GO here then please specifically mention what needs to be removed or create a proposal to remove. In can test that again local by `GO build` and rerun etcetera.

Reply:
@chkp-royl Thank you for your patience. It not clear to me what you ment. I removed this:
https://github.com/CheckPointSW/terraform-provider-checkpoint/pull/141/commits/df67e7f88394160024e73764265fdcf6092012ac 

`go build verify_policy` again and tested with same results.

Reply:
Great! Looks much better now. 
I have last comment on textual corrections so it will be more clear and correct for the user:
policy-package description & change "Verify" to "Verify policy .. "

Reply:
Good. I pushed those corrections also to the PR.
Let me know if that is alright now. Thanks again for your review.

Regards,
Roeland